.github/workflows/wheels-pure.yml

```diff
@@ -0,0 +1,192 @@
+
+        if [ "${versioneer_override}" == "" ]; then
+          versioneer_override="$(rapids-pip-wheel-version ${{ needs.wheel-epoch-timestamp.outputs.rapids_epoch_timestamp }})"
+        fi
+          
+        export RAPIDS_PY_WHEEL_VERSIONEER_OVERRIDE="${versioneer_override}"
+        python -m pip wheel -w ./dist ${{ inputs.package-dir }} --no-deps
+      shell: bash
+
+    - name: Upload wheels to downloads.rapids.ai
+      run: rapids-upload-wheels-to-s3 dist
+      env:
+        RAPIDS_PY_WHEEL_NAME: ${{ inputs.package-name }}
+        RAPIDS_BUILD_TYPE: ${{ inputs.build-type }}
+        RAPIDS_PY_VERSION: "3.8"
+        AWS_ACCESS_KEY_ID: ${{ secrets.RAPIDSAI_GHA_AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.RAPIDSAI_GHA_AWS_SECRET_ACCESS_KEY }}
+
+  wheel-test:
+    name: wheel test pure
+    runs-on: [self-hosted, linux, amd64, gpu-v100-495-1]
+    needs: wheel-build
+    container:
+      image: "rapidsai/citestwheel_x86_64"
+      env:
+        NVIDIA_VISIBLE_DEVICES: ${{ env.NVIDIA_VISIBLE_DEVICES }} # GPU jobs must set this container env variable
+        RAPIDS_PY_WHEEL_NAME: ${{ inputs.package-name }}
+        RAPIDS_CPYTHON_VERSION: "cp38"
+        RAPIDS_PY_VERSION: "3.8"
+        RAPIDS_BUILD_TYPE: ${{ inputs.build-type }}
+        AWS_ACCESS_KEY_ID: ${{ secrets.RAPIDSAI_GHA_AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.RAPIDSAI_GHA_AWS_SECRET_ACCESS_KEY }}
+        RAPIDS_BEFORE_TEST_COMMANDS_AMD64: ${{ inputs.test-before }}
+        CIBW_TEST_EXTRAS: ${{ inputs.test-extras }}
+        CIBW_TEST_COMMAND: ${{ inputs.test-unittest }}
+        PIP_EXTRA_INDEX_URL: "https://pypi.k8s.rapids.ai/simple"
+    steps:
+    - name: Run nvidia-smi to make sure GPU is working
+      run: nvidia-smi
+
+    - name: Install private index credentials in cibuildwheel container
+      run: printf 'machine pypi.k8s.rapids.ai\n\tlogin cibuildwheel\n\tpassword ${{ secrets.RAPIDSAI_PYPI_CI_PASSWORD }}\n' > ~/.netrc
+
+    - name: checkout code repo
+      uses: actions/checkout@v3
+      if: ${{ inputs.test-checkout-src == 'true' }}
+      with:
+        repository: ${{ inputs.repo }}
+        ref: ${{ inputs.sha }}
+        fetch-depth: 0 # unshallow fetch for setuptools-scm
+        persist-credentials: false
+
+    - name: Get repository information from build step
+      run: |
+        echo "RAPIDS_REPOSITORY=${{ needs.wheel-build.outputs.rapids_repository }}" >> "${GITHUB_ENV}"
+        echo "RAPIDS_SHA=${{ needs.wheel-build.outputs.rapids_sha }}" >> "${GITHUB_ENV}"
+        echo "RAPIDS_REF_NAME=${{ needs.wheel-build.outputs.rapids_ref_name }}" >> "${GITHUB_ENV}"
+        echo "RAPIDS_NIGHTLY_DATE=${{ needs.wheel-build.outputs.rapids_nightly_date }}" >> "${GITHUB_ENV}"
+
+    - name: Run citestwheel
+      run: /citestwheel.sh
+
+  wheel-publish:
+    name: wheel publish pure
+    needs: [wheel-build, wheel-test]
+    runs-on: [self-hosted, linux, amd64, cpu4]
```

Comment:
For simple publish steps like this, it might be better to use GitHub's hosted runners since:

- they're free
- they tend to start up a bit faster than our self-hosted nodes

That's what we did for our `conda-upload` step below:

- https://github.com/rapidsai/shared-action-workflows/blob/6bb1efd7f3b9589471e9fa1e20b879fb7fce8e28/.github/workflows/conda-upload-packages.yaml#L18